src/openrct2/drawing/Text.cpp

```diff
@@ -145,6 +145,29 @@ static void DrawTextEllipsisedCompat(rct_drawpixelinfo * dpi, sint32 x, sint32 y
     DrawText(dpi, x, y, &_legacyPaint, buffer);
 }
 
+static void DrawTextEllipsisedCompatNoFontReset(
```

Comment:
What does 'No Font Reset' mean?

Reply:
`DrawTextEllipsisedCompat` always resets `_legacyPaint.SpriteBase` to `FONT_SPRITE_BASE_MEDIUM`

This function uses the requested `gCurrentFontSpriteBase;`

I didn't want to introduce another parameter so introduced a separate function.